utils/src/main/scala/za/co/absa/enceladus/utils/validation/field/FieldValidator.scala

```diff
@@ -15,35 +15,25 @@
 
 package za.co.absa.enceladus.utils.validation.field
 
-import org.apache.spark.sql.types._
-import za.co.absa.enceladus.utils.validation.ValidationIssue
+import za.co.absa.enceladus.utils.types.TypedStructField
+import za.co.absa.enceladus.utils.validation.{ValidationError, ValidationIssue}
 
-trait FieldValidator {
-  def validateStructField(field: StructField): Seq[ValidationIssue]
-}
-
-object FieldValidator {
+import scala.util.{Failure, Success, Try}
 
-  /**
-    * Validates a StructField and returns the list of issues encountered
-    *
-    * @param field A field
-    * @return The list if validation issues encountered
-    */
-  def validate(field: StructField): Seq[ValidationIssue] = {
-    val validator = field.dataType match {
-      case _: DateType =>
-        Option(FieldValidatorDate)
-      case _: TimestampType =>
-        Option(FieldValidatorTimestamp)
-      case _: BooleanType =>
-        Option(FieldValidatorScalar)
-      case _: NumericType =>
-        Option(FieldValidatorScalar)
-      case _: StringType =>
-        Option(FieldValidatorScalar)
-      case _ => None
+class FieldValidator {
+  protected def tryToValidationIssues(tryValue: Try[Any]): Seq[ValidationIssue] = {
```

Comment:
This name is a bit weird to me

Reply:
Naming ala `stringToInt`, try is `Try` the noun.

Reply:
Document it please

Reply:
Done